### DISCUSSION TITLE
Not Able to Run Airflow Through Docker

### DISCUSSION THREAD

**Discussion by Nidesh-Arumugam** [apache/airflow#19348] [Q&A] [answered]
I have followed the official documentation from airflow for installing the latest version using Docker .
even after following the docs, i am not able to access the localhost:8080 as expected. My older version of airflow is still working good, but need to upgrade to newer version  in order to access some additional features.

<img width="1440" alt="Screenshot 2021-11-01 at 4 42 53 PM" src="https://user-images.githubusercontent.com/80671776/139663079-2dd201d7-147a-4ca5-86d1-491fc46e4d88.png">
<img width="1440" alt="Screenshot 2021-11-01 at 4 43 12 PM" src="https://user-images.githubusercontent.com/80671776/139663121-74b779a4-9693-48cb-89d4-b42c0fc81ef2.png">
<img width="1440" alt="Screenshot 2021-11-01 at 4 44 20 PM" src="https://user-images.githubusercontent.com/80671776/139663226-aeb7af1e-0e01-478e-ac73-6a5df87531d8.png">

**Comment by potiuk** [comment, marked as answer]
It's very likely you have not enough memory allocated for Docker: See "Before you begin" section https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html#before-you-begin - for details.

**Reply by Nidesh-Arumugam** [reply]
it says atleast 4 gb, i have 4.9g allocated for it

**Reply by potiuk** [reply]
Try more. It also says "ideally 8GB" - the 4GB is minimum requirements - depending on how much of other stuff you have it might be more. Did you actuallty try more memory to see if it helps? I think what I would do - If I were you - I'd try different thigns to see if it can be resolved. What did you try so far?

**Reply by potiuk** [reply]
Also you need ot take a look at the logs of docker-compose - it will likely tell you what is killing the containers (somethiong is) - but you should be able to analyse it and see what is killing it yourself, it's beyond of "airflow" itself, it's the deployment you have. The "quick-start" docker-compose is just that - quick starat, but it's clearly specifed that you need to know docker-compose and be able to be quite profficient with it in case you have any issues, modifications, customisation, deployments. Docker-compose is not "production-ready" - https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html#production-readiness so mostly investigation what's wrong with your deployment is mostly something that you need to do yourself and provide some more results of investigation. There are many variables in docker-compose deployments, and it's impossible to make it "just-works" for all cases. 

If you want something that should mostly "just-work" you should aim for https://airflow.apache.org/docs/helm-chart/stable/index.html as it has a lot ot "knobs" to turn and configuration parameters and it is much better supported by the community here. 

With docker-compose, you really need to be able to analyse it deeply and modify it to adjust for your use.

**Reply by Nidesh-Arumugam** [reply]
Thanks for the reply I tried them by increasing the Ram on the server, It works fine now. Got it solved few hours ago, Forgot to mention it here.

**Reply by potiuk** [reply]
OK. Good to know! Others might find it then and solve their problems faster :).

**Reply by ShawonSimon** [reply]
how did you increase the RAM?

**Comment by ShawonSimon** [comment]
how did you increase the RAM?
